_sub/security/aws-backup/main.tf

```diff
@@ -0,0 +1,188 @@
+# AWS Vault Creation
+resource "aws_backup_vault" "vault" {
+  name        = var.vault_name
+  kms_key_arn = var.deploy_kms_key ? aws_kms_key.this[0].arn : var.kms_key_arn
+  tags        = var.tags
+}
+
+# KMS Key for Encryption
+resource "aws_kms_key" "this" {
+  count               = var.deploy_kms_key ? 1 : 0
```

Comment:
count          = var.deploy ? 1 : 0
Needs to be added in addition to the condition